lib/clean-host.js

```diff
@@ -89,30 +76,45 @@ module.exports = function extractHostname(value) {
     url = '' + url;
   }
 
-  var needsTrimming = checkTrimmingNeeded(url);
-  if (needsTrimming) {
-    url = url.trim();
-  }
-
   var needsLowerCase = checkLowerCaseNeeded(url);
   if (needsLowerCase) {
     url = url.toLowerCase();
   }
 
+  var v6Match = url.match(ipv6LikeRE);
+  if (v6Match) {
+    return v6Match[1];
+  }
+
+  var needsTrimming = checkTrimmingNeeded(url);
+  if (needsTrimming) {
+    url = url.trim();
+  }
+
   // Try again after `url` has been transformed to lowercase and trimmed.
   if ((needsLowerCase || needsTrimming) && isValid(url)) {
     return trimTrailingDots(url);
   }
 
   // Proceed with heavier url parsing to extract the hostname.
   if (!hasPrefixRE.test(url)) {
-    url = '//' + url;
+    url = 'https://' + url;
```

Comment:
WHATWG URL does not support scheme-less URLs so I had to add a scheme to get it to parse.